software/23.Elixir/README.md

```diff
@@ -19,4 +19,143 @@ Let's help him build the chat app of his dreams ! :)~
+
+```
+
+Try running the app, and test if everything works. You should have a welcome page on `http://localhost:4000` if everything is ok.
+
+Once we know all is working, we will need models to handle messages, and users, go to the `ird_chat` directory and type the following commands:
+```bash
+mix phx.gen.channel Room # Type 'Y' once
+```
+
+You should have the following output:
+```bash
+* creating lib/irc_chat_web/channels/room_channel.ex
+* creating test/irc_chat_web/channels/room_channel_test.exs
+
+The default socket handler - IrcChatWeb.UserSocket - was not found.
+
+Do you want to create it? [Yn] y
+* creating lib/irc_chat_web/channels/user_socket.ex
+* creating assets/js/user_socket.js
+
+Add the socket handler to your `lib/irc_chat_web/endpoint.ex`, for example:
+
+    socket "/socket", IrcChatWeb.UserSocket,
+      websocket: true,
+      longpoll: false
+
+For the front-end integration, you need to import the `user_socket.js`
+in your `assets/js/app.js` file:
+
+    import "./user_socket.js"
+```
+
+The `room_channel.ex` file handles receiving/sending messages and the `room_channel_test.exs` tests basic interaction with the channel. 
+The last two lines prompt us to import `user_socket.js` in our `app.js` file, let's do it:
+
+```diff
+// We import the CSS which is extracted to its own file by esbuild.
+// Remove this line if you add a your own CSS build pipeline (e.g postcss).
+import "../css/app.css"
+
+// If you want to use Phoenix channels, run `mix help phx.gen.channel`
+// to get started and then uncomment the line below.
+-// import "./user_socket.js"
++ import "./user_socket.js"
+```
+
+We are also prompted to update `endpoint.ex`, to implement our channel. 
+The file is stored in `lib/irc_chat_web/endpoint.ex`.
+
+A `socket /live` statemtent should at line 13, you can put your socket justunder this one.
+
+```elixir
+defmodule IrcChatWeb.Endpoint do
+  use Phoenix.Endpoint, otp_app: :irc_chat
+
+  # The session will be stored in the cookie and signed,
+  # this means its contents can be read but not tampered with.
+  # Set :encryption_salt if you would also like to encrypt it.
+  @session_options [
+    store: :cookie,
+    key: "_irc_chat_key",
+    signing_salt: "1jRoL3v2"
+  ]
+
+  socket "/live", Phoenix.LiveView.Socket, websocket: [connect_info: [session: @session_options]]
+  socket "/socket", IrcChatWeb.UserSocket, websocket: true, longpoll: false
+
+...
+```
+
+## Step 3 -- Build the Socket Handler
+
+Next, we will need to create the piece of code to handle our socket. You can create a new file in `/lib/irc_chat_web`. It will define the `UserSocket` module, so I would advise you to name it `user_socket.ex`.
+
+Socket params are passed from the client and can be used to verify and authenticate a user. 
+You can find some apt documentation [here](https://hexdocs.pm/phoenix/Phoenix.Socket.html).
+
+> Dont forget to add a `channel` for your chat room ! 
+> (Google is your friend if you don't know what that means)
+
```

Comment:
I think at this step it could be interesting to list steps which need to be done and also show a final result because it's important to guide the student so he doesn’t get lost on what he has to do.